org.eclipse.scanning.points/src/org/eclipse/scanning/points/CompoundIterator.java

```diff
@@ -57,6 +59,9 @@ private IPosition createFirstPosition() throws GeneratorException {
 	@Override
 	public boolean hasNext() {
         next = getNext(); 
+        index++;
+        if (next!=null) next.setStepIndex(index);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00122)